Telegram/SourceFiles/core/basic_types.h

```diff
@@ -54,4 +54,4 @@ void as_const(const T&&) = delete;
 #include <cstdint>
 
 #define qsl(s) QStringLiteral(s)
-#define qstr(s) QLatin1String(s, sizeof(s) - 1)
+#define qstr(s) QLatin1String(s, s + sizeof(s))
```

Comment:
Looks like this change caused [a build failure](https://ci.appveyor.com/project/procxx/tdesktop/build/1.0.71-epouxvof#L505) (although I don't understand why); what's the purpose of it?

Reply:
It looks like the [corresponding signature for `QLatin1String` constructor](http://doc.qt.io/qt-5/qlatin1string.html#QLatin1String-2) was introduced in Qt 5.10, although we claim to support Qt 5.9+. [Our build server uses Qt 5.9](https://github.com/procxx/kepka/blob/2a2d4a260ec1c91aa73a0ccc670f5f5ca18fcd3c/appveyor.yml#L26), and that's likely the reason for the build failure.

Do you suggest us to migrate to Qt 5.10? If so, then please clearly state the reason.

Reply:
It's my fault, it was elegant solution but it is compiled only with Qt 5.10 and newer. I plan to reverted to old version.